### PR TITLE
Add piranha

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: recipes/piranha/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  number: 0
+  skip: true        # [win]
+
+requirements:
+  build:
+    - toolchain
+    - cmake
+    - gmp
+    - mpfr
+    - boost
```
Comment (scopatz) [inline recipes/piranha/meta.yaml]: Doesn't boost need to be pinned?

File: recipes/piranha/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% set version = "0.5" %}
+
+package:
+  name: piranha
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/piranha/meta.yaml]: Could change to `piranha-{{ version }}.tar.gz` to make better use of the source cache.

File: recipes/piranha/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - bzip2 1.0.*
+  run:
+    - gmp
+    - mpfr
+    - boost 1.62.*
+    - bzip2 1.0.*
+
+test:
+  commands:
+    - test -f $PREFIX/include/piranha/piranha.hpp
```
Comment (jakirkham) [inline recipes/piranha/meta.yaml]: Are there any libraries we can test for?

Comment (isuruf) [inline recipes/piranha/meta.yaml]: No, it's a header only library.

File: recipes/piranha/meta.yaml (hunk 2 above)
Comment (isuruf) [inline recipes/piranha/meta.yaml]: This doesn't have to match the name on `url`?

File: recipes/piranha/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/piranha/meta.yaml]: Apparently this is header only. So maybe pinning is unnecessary.

File: recipes/piranha/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  skip: true        # [win]
+
+requirements:
+  build:
+    - toolchain
+    - cmake
+    - gmp
+    - mpfr
+    - boost 1.62.*
+    - bzip2 1.0.*
```
Comment (jakirkham) [inline recipes/piranha/meta.yaml]: Given this is header only, do we actually need any of these (excepting `cmake`)?

Comment (isuruf) [inline recipes/piranha/meta.yaml]: Yes, the config file is generated looking at the dependencies.

File: recipes/piranha/meta.yaml (hunk 1 above)
Comment (isuruf) [inline recipes/piranha/meta.yaml]: Yes, maybe just a `>=` would suffice.

**Comment by isuruf** [comment]
ping

**Pull request merged**